sg_cmds.cpp: warning: use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]

This is a warning I get with ICX 2023.1.0 (which is based on very recent LLVM clang):

```
src/sgame/sg_cmds.cpp:3160:9: warning: use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]
        return Cmd_Sell_upgradeItem( ent, UP_LIGHTARMOUR ) |
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/sgame/sg_cmds.cpp:3160:9: note: cast one or both operands to int to silence this warning
src/sgame/sg_cmds.cpp:3160:9: warning: use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]
        return Cmd_Sell_upgradeItem( ent, UP_LIGHTARMOUR ) |
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/sgame/sg_cmds.cpp:3160:9: note: cast one or both operands to int to silence this warning
src/sgame/sg_cmds.cpp:3160:9: warning: use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]
        return Cmd_Sell_upgradeItem( ent, UP_LIGHTARMOUR ) |
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                           ||
src/sgame/sg_cmds.cpp:3160:9: note: cast one or both operands to int to silence this warning
src/sgame/sg_cmds.cpp:3254:10: warning: use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]
                return Cmd_Sell_weapons( ent ) | Cmd_Sell_upgrades( ent );
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                               ||
src/sgame/sg_cmds.cpp:3254:10: note: cast one or both operands to int to silence this warning
```

_Edit_: I also get this with AOCC 4.0.0 which is based on Clang 14.0.6.

--- a/src/sgame/sg_cmds.cpp
+++ b/src/sgame/sg_cmds.cpp
@@ -2497,10 +2497,11 @@ static bool Cmd_Sell_upgrades( gentity_t *ent )
 
 static bool Cmd_Sell_armour( gentity_t *ent )
 {
-	return Cmd_Sell_upgradeItem( ent, UP_LIGHTARMOUR ) |
-	       Cmd_Sell_upgradeItem( ent, UP_MEDIUMARMOUR ) |
-	       Cmd_Sell_upgradeItem( ent, UP_BATTLESUIT ) |
-	       Cmd_Sell_upgradeItem( ent, UP_RADAR );
+	bool sold = Cmd_Sell_upgradeItem( ent, UP_LIGHTARMOUR );
+	sold |= Cmd_Sell_upgradeItem( ent, UP_MEDIUMARMOUR );
+	sold |= Cmd_Sell_upgradeItem( ent, UP_BATTLESUIT );
+	sold |= Cmd_Sell_upgradeItem( ent, UP_RADAR );
+	return sold;
 }
 
 static bool Cmd_Sell_internal( gentity_t *ent, const char *s )
@@ -2591,7 +2592,9 @@ static bool Cmd_Sell_internal( gentity_t *ent, const char *s )
 	}
 	else if ( !Q_stricmp( s, "all" ) )
 	{
-		return Cmd_Sell_weapons( ent ) | Cmd_Sell_upgrades( ent );
+		bool sold = Cmd_Sell_weapons( ent );
+		sold |= Cmd_Sell_upgrades( ent );
+		return sold;
 	}
 	else
 	{

